From c952b58ea37a27cc5d5d66778b0e8e62dd29d43a Mon Sep 17 00:00:00 2001
From: Anton Kikin <a.kikin@tano-systems.com>
Date: Sun, 1 Mar 2020 05:40:34 +0300
Subject: [PATCH 2/2] Add support for notice level messages

Signed-off-by: Anton Kikin <a.kikin@tano-systems.com>
---
 syslog.lua | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/syslog.lua b/syslog.lua
index febda00..f224dfd 100644
--- a/syslog.lua
+++ b/syslog.lua
@@ -3,11 +3,12 @@ require"lsyslog"
 
 local convert =
 {
-	[logging.DEBUG] = lsyslog.LOG_DEBUG,
-	[logging.INFO]  = lsyslog.LOG_INFO,
-	[logging.WARN]  = lsyslog.LOG_WARNING,
-	[logging.ERROR] = lsyslog.LOG_ERR,
-	[logging.FATAL] = lsyslog.LOG_ALERT,
+	[logging.DEBUG]  = lsyslog.LOG_DEBUG,
+	[logging.INFO]   = lsyslog.LOG_INFO,
+	[logging.NOTICE] = lsyslog.LOG_NOTICE,
+	[logging.WARN]   = lsyslog.LOG_WARNING,
+	[logging.ERROR]  = lsyslog.LOG_ERR,
+	[logging.FATAL]  = lsyslog.LOG_ALERT,
 }
 
 function logging.syslog(ident, facility)
-- 
2.16.2.windows.1

