From a9950b4ea0cfb49e23905f122b2be74acb2b3ac9 Mon Sep 17 00:00:00 2001
From: Anton Kikin <a.kikin@tano-systems.com>
Date: Thu, 5 Jul 2018 03:40:23 +0300
Subject: [PATCH] test-utils: Use portable format for size_t in printf

---
 src/test-utils.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/test-utils.c b/src/test-utils.c
index 4241194..7825579 100644
--- a/src/test-utils.c
+++ b/src/test-utils.c
@@ -76,17 +76,17 @@ int main(int argc, char **argv)
 
 	if (encode) {
 		rv = lh_urlencode(encode, 0, &len, flags);
-		printf("length=%ld encoded=%s\n", len, rv);
+		printf("length=%zd encoded=%s\n", len, rv);
 		return 0;
 	}
 	else if (decode) {
 		rv = lh_urldecode(decode, 0, &len, flags);
-		printf("length=%ld decoded=%s\n", len, rv);
+		printf("length=%zd decoded=%s\n", len, rv);
 		return 0;
 	}
 	else if (hval && attr) {
 		rv = lh_header_attribute(hval, 0, attr, &len);
-		printf("length=%ld value=%s\n", len, rv);
+		printf("length=%zd value=%s\n", len, rv);
 		return 0;
 	}
 
-- 
2.7.4

