diff --git "a/arch/arm/mach-omap2/am33xx/board.c" "b/arch/arm/mach-omap2/am33xx/board.c"
index 62158a9..36f116f 100644
--- "a/arch/arm/mach-omap2/am33xx/board.c"
+++ "b/arch/arm/mach-omap2/am33xx/board.c"
@@ -532,14 +532,23 @@ void early_system_init(void)
 #ifdef CONFIG_SPL_BUILD
 void board_init_f(ulong dummy)
 {
+	int i;
+
 	hw_data_init();
 	early_system_init();
 	board_early_init_f();
 	sdram_init();
+
 	/* dram_init must store complete ramsize in gd->ram_size */
-	gd->ram_size = get_ram_size(
-			(void *)CONFIG_SYS_SDRAM_BASE,
-			CONFIG_MAX_RAM_BANK_SIZE);
+	for (i = 0; i < 1000; ++i) {
+		gd->ram_size = get_ram_size(
+				(void *)CONFIG_SYS_SDRAM_BASE,
+				CONFIG_MAX_RAM_BANK_SIZE);
+		if (gd->ram_size != 0) {
+			break;
+		}
+		udelay(25);
+	}
 }
 #endif
 
