From 09cf5f8e57174a8a915ea71bbb16905d55d382f8 Mon Sep 17 00:00:00 2001
From: Anton Kikin <a.kikin@tano-systems.com>
Date: Fri, 8 Feb 2019 22:38:50 +0300
Subject: [PATCH 01/11] service: allow to change service's current directory

Signed-off-by: Anton Kikin <a.kikin@tano-systems.com>
---
 service/instance.c | 19 +++++++++++++++++++
 service/instance.h |  1 +
 2 files changed, 20 insertions(+)

diff --git a/service/instance.c b/service/instance.c
index b0c9807..c3576d2 100644
--- a/service/instance.c
+++ b/service/instance.c
@@ -61,6 +61,7 @@ enum {
 	INSTANCE_ATTR_RELOADSIG,
 	INSTANCE_ATTR_TERMTIMEOUT,
 	INSTANCE_ATTR_FACILITY,
+	INSTANCE_ATTR_CHDIR,
 	__INSTANCE_ATTR_MAX
 };
 
@@ -88,6 +89,7 @@ static const struct blobmsg_policy instance_attr[__INSTANCE_ATTR_MAX] = {
 	[INSTANCE_ATTR_RELOADSIG] = { "reload_signal", BLOBMSG_TYPE_INT32 },
 	[INSTANCE_ATTR_TERMTIMEOUT] = { "term_timeout", BLOBMSG_TYPE_INT32 },
 	[INSTANCE_ATTR_FACILITY] = { "facility", BLOBMSG_TYPE_STRING },
+	[INSTANCE_ATTR_CHDIR] = { "chdir", BLOBMSG_TYPE_STRING },
 };
 
 enum {
@@ -392,6 +394,10 @@ instance_run(struct service_instance *in, int _stdout, int _stderr)
 		ERROR("failed to set user id %d: %m\n", in->uid);
 		exit(127);
 	}
+	if (in->chdir && chdir(in->chdir)) {
+		ERROR("failed to change directory to %s: %m\n", in->chdir);
+		exit(127);
+	}
 
 	execvp(argv[0], argv);
 	exit(127);
@@ -699,6 +705,9 @@ instance_config_changed(struct service_instance *in, struct service_instance *in
 	if (!blobmsg_list_equal(&in->errors, &in_new->errors))
 		return true;
 
+	if (string_changed(in->chdir, in_new->chdir))
+		return true;
+
 	return false;
 }
 
@@ -977,6 +986,12 @@ instance_config_parse(struct service_instance *in)
 	if (tb[INSTANCE_ATTR_STDERR] && blobmsg_get_bool(tb[INSTANCE_ATTR_STDERR]))
 		in->_stderr.fd.fd = -1;
 
+	if (tb[INSTANCE_ATTR_CHDIR]) {
+		char *chdir = blobmsg_get_string(tb[INSTANCE_ATTR_CHDIR]);
+		if (chdir)
+			in->chdir = chdir;
+	}
+
 	instance_fill_any(&in->data, tb[INSTANCE_ATTR_DATA]);
 
 	if (!instance_fill_array(&in->env, tb[INSTANCE_ATTR_ENV], NULL, false))
@@ -1041,6 +1056,7 @@ instance_config_move(struct service_instance *in, struct service_instance *in_sr
 	in->seccomp = in_src->seccomp;
 	in->node.avl.key = in_src->node.avl.key;
 	in->syslog_facility = in_src->syslog_facility;
+	in->chdir = in_src->chdir;
 
 	free(in->config);
 	in->config = in_src->config;
@@ -1214,5 +1230,8 @@ void instance_dump(struct blob_buf *b, struct service_instance *in, int verbose)
 	if (verbose && in->trigger)
 		blobmsg_add_blob(b, in->trigger);
 
+	if (in->chdir)
+		blobmsg_add_string(b, "chdir", in->chdir);
+
 	blobmsg_close_table(b, i);
 }
diff --git a/service/instance.h b/service/instance.h
index 42cc4be..ff7cda2 100644
--- a/service/instance.h
+++ b/service/instance.h
@@ -62,6 +62,7 @@ struct service_instance {
 	char *seccomp;
 	char *pidfile;
 	int syslog_facility;
+	char *chdir;
 
 	uint32_t term_timeout;
 	uint32_t respawn_timeout;
-- 
2.16.2.windows.1

