From 2237c51227e60a7fe91c6a28b19a14d1236cc74c Mon Sep 17 00:00:00 2001
From: Anton Kikin <a.kikin@tano-systems.com>
Date: Sat, 22 Jun 2019 06:02:50 +0300
Subject: [PATCH 05/15] service: Add SCHED_IDLE scheduler policy support

Signed-off-by: Anton Kikin <a.kikin@tano-systems.com>
---
 service/instance.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/service/instance.c b/service/instance.c
index 5676ebe..4c10972 100644
--- a/service/instance.c
+++ b/service/instance.c
@@ -201,6 +201,7 @@ static const struct policy_info policies[] = {
 	{ "SCHED_RR", SCHED_RR },
 	{ "SCHED_BATCH", SCHED_BATCH },
 	{ "SCHED_DEADLINE", SCHED_DEADLINE },
+	{ "SCHED_IDLE", SCHED_IDLE },
 	{ NULL, 0 }
 };
 
@@ -480,6 +481,9 @@ instance_run(struct service_instance *in, int _stdout, int _stderr)
 				in->srv->name, in->name, in->period);
 			break;
 
+		case SCHED_IDLE:
+			break;
+
 		default:
 			break;
 	}
@@ -1871,6 +1875,9 @@ void instance_dump(struct blob_buf *b, struct service_instance *in, int verbose)
 			blobmsg_add_u64(b, "period", in->period);
 			break;
 
+		case SCHED_IDLE:
+			break;
+
 		default:
 			break;
 	}
-- 
2.28.0.windows.1

