From 1d123a40e51e5b2fca9815dbf07f62c619534741 Mon Sep 17 00:00:00 2001
From: Anton Kikin <a.kikin@tano-systems.com>
Date: Fri, 6 Mar 2020 03:13:10 +0300
Subject: [PATCH 11/24] system: Add tm_gmtoff to system.info ubus call

Signed-off-by: Anton Kikin <a.kikin@tano-systems.com>
---
 system.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/system.c b/system.c
index c208e3e..4319831 100644
--- a/system.c
+++ b/system.c
@@ -361,6 +361,7 @@ static int system_info(struct ubus_context *ctx, struct ubus_object *obj,
 	blob_buf_init(&b, 0);
 
 	blobmsg_add_u32(&b, "localtime", now + tm->tm_gmtoff);
+	blobmsg_add_u32(&b, "localtime_gmtoff", tm->tm_gmtoff);
 
 #ifdef linux
 	blobmsg_add_u32(&b, "uptime",    info.uptime);
-- 
2.28.0.windows.1

