From ef05a1277201e10c9d93816cb90d3e0462d33e15 Mon Sep 17 00:00:00 2001
From: Anton Kikin <a.kikin@tano-systems.com>
Date: Mon, 6 Apr 2020 06:17:15 +0300
Subject: [PATCH 12/24] system: Handle DISTRIB_TIMESTAMP in
 /etc/openwrt_release

Signed-off-by: Anton Kikin <a.kikin@tano-systems.com>
---
 system.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/system.c b/system.c
index bbff932..b3261b7 100644
--- a/system.c
+++ b/system.c
@@ -272,6 +272,8 @@ static int system_board(struct ubus_context *ctx, struct ubus_object *obj,
 				key = "target";
 			else if (!strcasecmp(key, "DISTRIB_DESCRIPTION"))
 				key = "description";
+			else if (!strcasecmp(key, "DISTRIB_TIMESTAMP"))
+				key = "timestamp";
 			else
 				continue;
 
-- 
2.33.0.windows.2

