From f6da8fff6766be9548675e38381c8c2fd68de33b Mon Sep 17 00:00:00 2001
From: Anton Kikin <a.kikin@tano-systems.com>
Date: Wed, 11 Nov 2020 06:58:50 +0300
Subject: [PATCH 13/24] uxc: Fix compiler warnings

Signed-off-by: Anton Kikin <a.kikin@tano-systems.com>
---
 uxc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/uxc.c b/uxc.c
index b603086..e9ea32d 100644
--- a/uxc.c
+++ b/uxc.c
@@ -1250,7 +1250,7 @@ int main(int argc, char **argv)
 				break;
 
 			case 'v':
-				verbose = true;
+//				verbose = true;
 				break;
 
 			case 'V':
-- 
2.28.0.windows.1

