From f661f76f68fe8722d19222e9eafdaf3f5e559f4a Mon Sep 17 00:00:00 2001
From: Anton Kikin <a.kikin@tano-systems.com>
Date: Wed, 11 Nov 2020 06:58:50 +0300
Subject: [PATCH 13/24] uxc: Fix compiler warnings

Signed-off-by: Anton Kikin <a.kikin@tano-systems.com>
---
 uxc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/uxc.c b/uxc.c
index 332b91d..9d6eab5 100644
--- a/uxc.c
+++ b/uxc.c
@@ -1446,7 +1446,7 @@ int main(int argc, char **argv)
 				break;
 
 			case 'v':
-				verbose = true;
+//				verbose = true;
 				break;
 
 			case 'V':
-- 
2.34.1.windows.1

