From 0c3e31166d713ea9d06053243bbad564b7ec75a9 Mon Sep 17 00:00:00 2001
From: Anton Kikin <a.kikin@tano-systems.com>
Date: Wed, 11 Nov 2020 06:58:50 +0300
Subject: [PATCH 13/21] uxc: Fix compiler warnings

Signed-off-by: Anton Kikin <a.kikin@tano-systems.com>
---
 uxc.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/uxc.c b/uxc.c
index eb40eb2..54dd7c3 100644
--- a/uxc.c
+++ b/uxc.c
@@ -724,7 +724,7 @@ int main(int argc, char **argv)
 	char *pidfile = NULL;
 	bool autostart = false;
 	bool force = false;
-	bool verbose = false;
+//	bool verbose = false;
 	int signal = SIGTERM;
 	int c;
 
@@ -775,7 +775,7 @@ int main(int argc, char **argv)
 				break;
 
 			case 'v':
-				verbose = true;
+//				verbose = true;
 				break;
 
 			case 'V':
-- 
2.16.2.windows.1

