From 12fdc6f92173d844a598bedf2bfe5c7ce7bf2c81 Mon Sep 17 00:00:00 2001
From: Anton Kikin <a.kikin@tano-systems.com>
Date: Wed, 11 Nov 2020 06:58:50 +0300
Subject: [PATCH 13/17] uxc: Fix compiler warnings

Signed-off-by: Anton Kikin <a.kikin@tano-systems.com>
---
 uxc.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/uxc.c b/uxc.c
index 3c6942a..e9d23e5 100644
--- a/uxc.c
+++ b/uxc.c
@@ -690,8 +690,8 @@ int main(int argc, char **argv)
 	char *bundle = NULL;
 	char *pidfile = NULL;
 	bool autostart = false;
-	bool force = false;
-	bool verbose = false;
+//	bool force = false;
+//	bool verbose = false;
 	int signal = SIGTERM;
 	int c;
 
@@ -734,7 +734,7 @@ int main(int argc, char **argv)
 				break;
 
 			case 'f':
-				force = true;
+//				force = true;
 				break;
 
 			case 'p':
@@ -742,7 +742,7 @@ int main(int argc, char **argv)
 				break;
 
 			case 'v':
-				verbose = true;
+//				verbose = true;
 				break;
 
 			case 'V':
-- 
2.16.2.windows.1

