From 7138147a04c2881057f1c344d1e991e8675ccd26 Mon Sep 17 00:00:00 2001
From: Anton Kikin <a.kikin@tano-systems.com>
Date: Wed, 11 Nov 2020 07:19:01 +0300
Subject: [PATCH 15/17] jail: Fix compiler warnings

Signed-off-by: Anton Kikin <a.kikin@tano-systems.com>
---
 jail/cgroups.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/jail/cgroups.c b/jail/cgroups.c
index 740997e..0e14782 100644
--- a/jail/cgroups.c
+++ b/jail/cgroups.c
@@ -639,7 +639,7 @@ static int parseOCIlinuxcgroups_legacy_memory(struct blob_attr *msg)
 {
 	struct blob_attr *tb[__OCI_LINUX_CGROUPS_MEMORY_MAX];
 	char tmp[32] = { 0 };
-	int64_t limit, swap, reservation;
+	int64_t limit = -1, swap, reservation;
 
 	blobmsg_parse(oci_linux_cgroups_memory_policy, __OCI_LINUX_CGROUPS_MEMORY_MAX, tb, blobmsg_data(msg), blobmsg_len(msg));
 
-- 
2.16.2.windows.1

