From 524d30c6a30ada7deb3cf3519a247ea1e39864eb Mon Sep 17 00:00:00 2001
From: Anton Kikin <a.kikin@tano-systems.com>
Date: Thu, 17 Dec 2020 21:15:32 +0300
Subject: [PATCH 19/24] rcS: Display only script name in psplash instead of
 full path

Signed-off-by: Anton Kikin <a.kikin@tano-systems.com>
---
 rcS.c | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/rcS.c b/rcS.c
index 60cc4e7..fd55af3 100644
--- a/rcS.c
+++ b/rcS.c
@@ -23,6 +23,10 @@
 #include <stdio.h>
 #include <unistd.h>
 
+#if defined(PSPLASH_SUPPORT)
+#include <ctype.h> /* isdigit() */
+#endif
+
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
@@ -128,7 +132,14 @@ static void q_initd_run(struct runqueue *q, struct runqueue_task *t)
 
 #if defined(PSPLASH_SUPPORT) && defined(PSPLASH_SCRIPT_MSG)
 	if (s->is_boot)
-		psplash_update_initd_script(s->file);
+	{
+		if (!strncmp(s->file, "/etc/rc.d/S", 11) &&
+		     isdigit(s->file[11]) &&
+		     isdigit(s->file[12]))
+			psplash_update_initd_script(&s->file[13]);
+		else
+			psplash_update_initd_script(s->file);
+	}
 #endif
 
 	clock_gettime(CLOCK_MONOTONIC_RAW, &s->ts_start);
-- 
2.34.1.windows.1

