From dd03d51aeee0ba46a022c5053d5e2c367f743355 Mon Sep 17 00:00:00 2001
From: Anton Kikin <a.kikin@tano-systems.com>
Date: Thu, 11 Jul 2019 04:50:21 +0300
Subject: [PATCH 23/24] procd: Add shared and slab to memory table

Signed-off-by: Anton Kikin <a.kikin@tano-systems.com>
---
 system.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/system.c b/system.c
index 5391ff1..b8626cd 100644
--- a/system.c
+++ b/system.c
@@ -334,6 +334,7 @@ static int system_info(struct ubus_context *ctx, struct ubus_object *obj,
 	char line[256];
 	char *key, *val;
 	unsigned long long available, cached;
+	unsigned long long shmem, slab;
 	FILE *f;
 	int i;
 	struct statvfs s;
@@ -351,6 +352,8 @@ static int system_info(struct ubus_context *ctx, struct ubus_object *obj,
 	/* if linux < 3.14 MemAvailable is not in meminfo */
 	available = 0;
 	cached = 0;
+	shmem = 0;
+	slab = 0;
 
 	while (fgets(line, sizeof(line), f))
 	{
@@ -364,6 +367,10 @@ static int system_info(struct ubus_context *ctx, struct ubus_object *obj,
 			available = 1024 * atoll(val);
 		else if (!strcasecmp(key, "Cached"))
 			cached = 1024 * atoll(val);
+		else if (!strcasecmp(key, "Shmem"))
+			shmem = 1024 * atoll(val);
+		else if (!strcasecmp(key, "Slab"))
+			slab = 1024 * atoll(val);
 	}
 
 	fclose(f);
@@ -399,6 +406,8 @@ static int system_info(struct ubus_context *ctx, struct ubus_object *obj,
 			(uint64_t)info.mem_unit * (uint64_t)info.bufferram);
 	blobmsg_add_u64(&b, "available", available);
 	blobmsg_add_u64(&b, "cached", cached);
+	blobmsg_add_u64(&b, "shmem", shmem);
+	blobmsg_add_u64(&b, "slab", slab);
 	blobmsg_close_table(&b, c);
 
 	for (i = 0; i < sizeof(fslist) / sizeof(fslist[0]); i += 2) {
-- 
2.34.1.windows.1

