From 970667813e214645cddc526b6a90df989af90b15 Mon Sep 17 00:00:00 2001
From: Anton Kikin <a.kikin@tano-systems.com>
Date: Tue, 15 Jun 2021 01:29:27 +0300
Subject: [PATCH 4/7] use-libnl3-instead-of-libnl-tiny

---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 5d4855d..fd5fb1a 100644
--- a/Makefile
+++ b/Makefile
@@ -34,8 +34,8 @@ endif
 
 ifneq ($(filter nl80211,$(IWINFO_BACKENDS)),)
 	IWINFO_CFLAGS      += -DUSE_NL80211
-	IWINFO_CLI_LDFLAGS += -lnl-tiny
-	IWINFO_LIB_LDFLAGS += -lnl-tiny
+	IWINFO_CLI_LDFLAGS += -lnl-3 -lnl-genl-3
+	IWINFO_LIB_LDFLAGS += -lnl-3 -lnl-genl-3
 	IWINFO_LIB_OBJ     += iwinfo_nl80211.o
 endif
 
-- 
2.34.1.windows.1

