From: Anton Kikin <a.kikin@tano-systems.com>
Date: Thu, 13 Oct 2018 00:06:24 +0300
Subject: [PATCH] Fix mounted filesystems info

Fixed mounted filesystems size and filesystem type output.

Signed-off-by: Anton Kikin <a.kikin@tano-systems.com>
---

diff --git a/modules/luci-mod-system/luasrc/model/cbi/admin_system/fstab.lua b/modules/luci-mod-system/luasrc/model/cbi/admin_system/fstab.lua
index 02fe9cc05..033b6d776 100644
--- a/modules/luci-mod-system/luasrc/model/cbi/admin_system/fstab.lua
+++ b/modules/luci-mod-system/luasrc/model/cbi/admin_system/fstab.lua
@@ -157,7 +157,7 @@ dev.cfgvalue = function(self, section)
 	local v, e
 
 	v = m.uci:get("fstab", section, "uuid")
-	e = v and devices[v:lower()]
+	e = v and devices[v]
 	if v and e and e.size then
 		return "UUID: %s (%s, %d MB)" %{ tp.pcdata(v), e.dev, e.size }
 	elseif v and e then
@@ -201,7 +201,7 @@ fs.cfgvalue = function(self, section)
 	local v, e
 
 	v = m.uci:get("fstab", section, "uuid")
-	v = v and v:lower() or m.uci:get("fstab", section, "label")
+	v = v or m.uci:get("fstab", section, "label")
 	v = v or m.uci:get("fstab", section, "device")
 
 	e = v and devices[v]
