From 25483cb3efc5ea8970b098c17c31a761c0944445 Mon Sep 17 00:00:00 2001
From: Anton Kikin <a.kikin@tano-systems.com>
Date: Fri, 27 Dec 2019 04:06:44 +0300
Subject: [PATCH 10/17] libfstools: Fix 64-bit compilation issues

Signed-off-by: Anton Kikin <a.kikin@tano-systems.com>
---
 libfstools/common.c | 4 ++--
 libfstools/common.h | 1 +
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/libfstools/common.c b/libfstools/common.c
index 4118c53..0bf981a 100644
--- a/libfstools/common.c
+++ b/libfstools/common.c
@@ -137,14 +137,14 @@ int block_volume_format(struct volume *v, uint64_t offset, const char *bdev)
 		ULOG_INFO("overlay filesystem in %s has not been formatted yet\n", v->blk);
 #ifdef OVL_F2FS_ENABLE
 		if (use_f2fs(v, offset, bdev)) {
-			ULOG_INFO("creating f2fs overlay filesystem (%s, offset %llu)...\n",
+			ULOG_INFO("creating f2fs overlay filesystem (%s, offset %"PRIu64")...\n",
 				v->blk, offset);
 			snprintf(str, sizeof(str), "mkfs.f2fs -q -l %s %s", get_overlay_partition(), v->blk);
 		}
 		else
 #endif
 		{
-			ULOG_INFO("creating ext4 overlay filesystem (%s, offset %llu)...\n",
+			ULOG_INFO("creating ext4 overlay filesystem (%s, offset %"PRIu64")...\n",
 				v->blk, offset);
 			snprintf(str, sizeof(str), "mkfs.ext4 -q -L %s %s", get_overlay_partition(), v->blk);
 		}
diff --git a/libfstools/common.h b/libfstools/common.h
index cde40eb..12384ba 100644
--- a/libfstools/common.h
+++ b/libfstools/common.h
@@ -16,6 +16,7 @@
 #include <unistd.h>
 #include <libgen.h>
 #include <fnmatch.h>
+#include <inttypes.h>
 
 #include "libfstools.h"
 #include "volume.h"
-- 
2.28.0.windows.1

