From 33c693ae6d734c431e4b803ef52b42a70ffcb5ba Mon Sep 17 00:00:00 2001
From: Anton Kikin <a.kikin@tano-systems.com>
Date: Wed, 25 Nov 2020 03:26:44 +0300
Subject: [PATCH 12/17] block: config_try_load: Log warning instead of error on
 failed attempt

Signed-off-by: Anton Kikin <a.kikin@tano-systems.com>
---
 block.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/block.c b/block.c
index 4d5e2b4..2033523 100644
--- a/block.c
+++ b/block.c
@@ -420,7 +420,7 @@ static struct uci_package * config_try_load(struct uci_context *ctx, char *path)
 
 	if (uci_load(ctx, file, &pkg)) {
 		uci_get_errorstr(ctx, &err, file);
-		ULOG_ERR("unable to load configuration (%s)\n", err);
+		ULOG_WARN("unable to load configuration (%s)\n", err);
 
 		free(err);
 		return NULL;
-- 
2.34.1.windows.1

