From e65aa9e84ff29b94b6ac4699a8a212559ccfa461 Mon Sep 17 00:00:00 2001
From: Anton Kikin <a.kikin@tano-systems.com>
Date: Tue, 9 Feb 2021 02:42:25 +0300
Subject: [PATCH 15/17] rootdisk: Increase maximum partitions number to 16

Signed-off-by: Anton Kikin <a.kikin@tano-systems.com>
---
 libfstools/rootdisk.c | 17 ++++++++++-------
 1 file changed, 10 insertions(+), 7 deletions(-)

diff --git a/libfstools/rootdisk.c b/libfstools/rootdisk.c
index 4c8d58e..8be91ab 100644
--- a/libfstools/rootdisk.c
+++ b/libfstools/rootdisk.c
@@ -52,16 +52,18 @@ struct blkdev_diskpart_info {
 	char dev[64];   /* partition device */
 };
 
+#define MAX_PARTITIONS 16
+
 struct blkdev_disk_info {
 	/* partitions */
-	struct blkdev_diskpart_info partition[4];
+	struct blkdev_diskpart_info partition[MAX_PARTITIONS];
 
 	const char *dev;         /* device */
 	const char *devname;     /* device name without "/dev/" prefix */
 	uint64_t hw_sector_size; /* sector size */
 	uint64_t size;           /* disk size in bytes */
 	int partitioned;         /* device has partitions? */
-	int rootpart;            /* rootfs partition number (1...4) */
+	int rootpart;            /* rootfs partition number (1...MAX_PARTITIONS) */
 	int ro;                  /* entire disk has ro flag setted? */
 };
 
@@ -147,7 +149,8 @@ static int parse_cmdline()
 		ovl_partition = (int)strtol(buf, NULL, 0);
 		if ((ovl_partition < 1) || (ovl_partition > MAX_PARTITIONS)) {
 			ovl_partition = 0;
-			ULOG_ERR("wrong 'ovl-partition' value (should be in the range from 1 to 4)\n");
+			ULOG_ERR("wrong 'ovl-partition' value (should be in the range from 1 to %d)\n",
+				MAX_PARTITIONS);
 		} else {
 			ULOG_INFO("try to use ovl-partition %d for overlay filesystem\n", ovl_partition);
 		}
@@ -230,7 +233,7 @@ static int rootdisk_scan_partitions(
 			if (read_u64_from_file(buf, "partition", &partition) != 0)
 				continue; /* device is not partition */
 
-			if ((partition < 1) || (partition > 4))
+			if ((partition < 1) || (partition > MAX_PARTITIONS))
 				continue; /* invalid partition number */
 
 			part = &info->partition[partition - 1];
@@ -363,7 +366,7 @@ static int blkdev_get_disk_info(
 		return -1;
 
 	/* Output partition table */
-	for (part = 0; part < 4; part++) {
+	for (part = 0; part < MAX_PARTITIONS; part++) {
 		if (info->partition[part].partition != (part + 1))
 			continue;
 
@@ -396,7 +399,7 @@ static int rootdisk_volume_find_partition(
 {
 	int part;
 
-	for (part = 0; part < 4; part++) {
+	for (part = 0; part < MAX_PARTITIONS; part++) {
 		if (!rootdisk_info.partition[part].partition)
 			continue;
 
@@ -507,7 +510,7 @@ static struct volume *rootdisk_volume_find(char *name)
 					p->no_overlay_create = 1;
 				} else {
 					/* try to put overlayfs at end of the partitions on disk */
-					for (part = rootdisk_info.rootpart; part <= 4; part++)
+					for (part = rootdisk_info.rootpart; part <= MAX_PARTITIONS; part++)
 					{
 						if (rootdisk_info.partition[part - 1].partition != part)
 							continue;
-- 
2.28.0.windows.1

