From 787d4f81c8df66ca64c0baf0531e57d213515f41 Mon Sep 17 00:00:00 2001
From: Anton Kikin <a.kikin@tano-systems.com>
Date: Thu, 11 Jun 2020 02:53:02 +0300
Subject: [PATCH] Add support for 5.x kernels

Signed-off-by: Anton Kikin <a.kikin@tano-systems.com>
---
 master/cdev.c | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/master/cdev.c b/master/cdev.c
index 52b1e01..9583493 100644
--- a/master/cdev.c
+++ b/master/cdev.c
@@ -62,7 +62,11 @@ static int eccdev_mmap(struct file *, struct vm_area_struct *);
 #define PAGE_FAULT_VERSION KERNEL_VERSION(2, 6, 23)
 
 #if LINUX_VERSION_CODE >= PAGE_FAULT_VERSION
+#if LINUX_VERSION_CODE > KERNEL_VERSION(4, 16, 0)
+static vm_fault_t eccdev_vma_fault(struct vm_fault *);
+#else
 static int eccdev_vma_fault(struct vm_fault *);
+#endif
 #else
 static struct page *eccdev_vma_nopage(
         struct vm_area_struct *, unsigned long, int *);
@@ -252,7 +256,12 @@ int eccdev_mmap(
  *
  * \return Zero on success, otherwise a negative error code.
  */
-static int eccdev_vma_fault(
+#if LINUX_VERSION_CODE > KERNEL_VERSION(4, 16, 0)
+static vm_fault_t
+#else
+static int
+#endif
+eccdev_vma_fault(
         struct vm_fault *vmf /**< Fault data. */
         )
 {
-- 
2.16.2.windows.1

