diff --git a/sys.c b/sys.c
index 60ff5d6..39dbee8 100644
--- a/sys.c
+++ b/sys.c
@@ -88,7 +88,7 @@ rpc_cgi_password_set(struct ubus_context *ctx, struct ubus_object *obj,
 	ssize_t n;
 	int ret;
 	const char *const passwd = "/bin/passwd";
-	const struct timespec req = {0, 100 * 1000 * 1000};
+	const struct timespec treq = {0, 100 * 1000 * 1000};
 
 	blobmsg_parse(rpc_password_policy, __RPC_P_MAX, tb,
 	              blob_data(msg), blob_len(msg));
@@ -146,7 +146,7 @@ rpc_cgi_password_set(struct ubus_context *ctx, struct ubus_object *obj,
 		if (n < 0)
 			return rpc_errno_status();
 
-		nanosleep(&req, NULL);
+		nanosleep(&treq, NULL);
 
 		n = write(fds[1], blobmsg_data(tb[RPC_P_PASSWORD]),
 		              blobmsg_data_len(tb[RPC_P_PASSWORD]) - 1);
