From 79dd412baa3c2040e7651c6fd586a8d968f4a6fb Mon Sep 17 00:00:00 2001
From: Anton Kikin <a.kikin@tano-systems.com>
Date: Mon, 4 Feb 2019 19:20:28 +0300
Subject: [PATCH 1/2] Change configuration paths

Signed-off-by: Anton Kikin <a.kikin@tano-systems.com>
---
 libopkg/opkg_conf.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/libopkg/opkg_conf.h b/libopkg/opkg_conf.h
index 0f65cab..ae48870 100644
--- a/libopkg/opkg_conf.h
+++ b/libopkg/opkg_conf.h
@@ -35,12 +35,12 @@ extern "C" {
 #define OPKG_CONF_TMP_DIR_SUFFIX "opkg-XXXXXX"
 #define OPKG_BACKUP_SUFFIX "-opkg.backup"
 
-#define OPKG_CONF_DEFAULT_LISTS_DIR     VARDIR "/lib/opkg/lists"
-#define OPKG_CONF_DEFAULT_INFO_DIR      VARDIR "/lib/opkg/info"
-#define OPKG_CONF_DEFAULT_STATUS_FILE   VARDIR "/lib/opkg/status"
+#define OPKG_CONF_DEFAULT_LISTS_DIR     "/usr/lib/opkg/lists"
+#define OPKG_CONF_DEFAULT_INFO_DIR      "/usr/lib/opkg/info"
+#define OPKG_CONF_DEFAULT_STATUS_FILE   "/usr/lib/opkg/status"
 #define OPKG_CONF_DEFAULT_CACHE_DIR     VARDIR "/cache/opkg"
 #define OPKG_CONF_DEFAULT_CONF_FILE_DIR SYSCONFDIR "/opkg"
-#define OPKG_CONF_DEFAULT_LOCK_FILE     "/run/opkg.lock"
+#define OPKG_CONF_DEFAULT_LOCK_FILE     "/var/run/opkg.lock"
 
 /* In case the config file defines no dest */
 #define OPKG_CONF_DEFAULT_DEST_NAME "root"
-- 
2.16.2.windows.1

