From 3eff5f2a2e33b3d05d29d2a31390983dda9c8b65 Mon Sep 17 00:00:00 2001
From: Anton Kikin <a.kikin@tano-systems.com>
Date: Thu, 24 Jun 2021 07:29:22 +0300
Subject: [PATCH 1/2] system-linux: Disable IFLA_VXLAN_GBP support for kernels
 < 4.0.0

Signed-off-by: Anton Kikin <a.kikin@tano-systems.com>
---
 system-linux.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/system-linux.c b/system-linux.c
index 45a9efb..d73e818 100644
--- a/system-linux.c
+++ b/system-linux.c
@@ -3691,9 +3691,11 @@ static void system_vxlan_map_bool_attr(struct nl_msg *msg, struct blob_attr **tb
 		if (invert)
 			val = !val;
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,0,0)
 		if ((attrtype == IFLA_VXLAN_GBP) && val)
 			nla_put_flag(msg, attrtype);
 		else 
+#endif
 			nla_put_u8(msg, attrtype, val);
 
 	}
@@ -3844,7 +3846,9 @@ static int system_add_vxlan(const char *name, const unsigned int link, struct bl
 	system_vxlan_map_bool_attr(msg, tb_data, IFLA_VXLAN_PROXY , VXLAN_DATA_ATTR_PROXY, false);
 	system_vxlan_map_bool_attr(msg, tb_data, IFLA_VXLAN_L2MISS , VXLAN_DATA_ATTR_L2MISS, false);
 	system_vxlan_map_bool_attr(msg, tb_data, IFLA_VXLAN_L3MISS , VXLAN_DATA_ATTR_L3MISS, false);
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,0,0)
 	system_vxlan_map_bool_attr(msg, tb_data, IFLA_VXLAN_GBP , VXLAN_DATA_ATTR_GBP, false);
+#endif
 
 	if ((cur = tb_data[VXLAN_DATA_ATTR_AGEING])) {
 		uint32_t ageing = blobmsg_get_u32(cur);
-- 
2.34.1.windows.1

