From d00ce092170311896498982adb01ff7516921afd Mon Sep 17 00:00:00 2001
From: Anton Kikin <a.kikin@tano-systems.com>
Date: Fri, 11 Dec 2020 23:15:54 +0300
Subject: [PATCH] Fix errors for clang

Signed-off-by: Anton Kikin <a.kikin@tano-systems.com>
---
 interface.c   | 2 ++
 proto-shell.c | 3 +++
 2 files changed, 5 insertions(+)

diff --git a/interface.c b/interface.c
index ccae915..48e74d4 100644
--- a/interface.c
+++ b/interface.c
@@ -246,6 +246,7 @@ interface_event(struct interface *iface, enum interface_event ev)
 		interface_error_flush(iface);
 		adev = iface->l3_dev.dev;
 		/* fall through */
+		fallthrough;
 	case IFEV_DOWN:
 	case IFEV_UP_FAILED:
 		alias_notify_device(iface->name, adev);
@@ -319,6 +320,7 @@ __interface_set_down(struct interface *iface, bool force)
 	case IFS_DOWN:
 		if (iface->main_dev.dev)
 			device_release(&iface->main_dev);
+		fallthrough;
 	case IFS_TEARDOWN:
 	default:
 		break;
diff --git a/proto-shell.c b/proto-shell.c
index 2662216..f68aebf 100644
--- a/proto-shell.c
+++ b/proto-shell.c
@@ -209,6 +209,7 @@ proto_shell_handler(struct interface_proto_state *proto,
 			}
 		/* if no script task is running */
 		/* fall through */
+			fallthrough;
 		case S_IDLE:
 			action = "teardown";
 			state->renew_pending = false;
@@ -293,6 +294,7 @@ proto_shell_task_finish(struct proto_shell_state *state,
 		if (task == &state->proto_task)
 			state->proto.proto_event(&state->proto, IFPEV_LINK_LOST);
 		/* fall through */
+		fallthrough;
 	case S_SETUP:
 		if (task == &state->proto_task)
 			proto_shell_handler(&state->proto, PROTO_CMD_TEARDOWN,
@@ -765,6 +767,7 @@ proto_shell_setup_failed(struct proto_shell_state *state)
 	case S_IDLE:
 		state->proto.proto_event(&state->proto, IFPEV_LINK_LOST);
 		/* fall through */
+		fallthrough;
 	case S_SETUP:
 		proto_shell_handler(&state->proto, PROTO_CMD_TEARDOWN, false);
 		break;
-- 
2.16.2.windows.1

