From 71f7bc460d6aa01d7e3824cd87192b9d47f5e571 Mon Sep 17 00:00:00 2001
From: Anton Kikin <a.kikin@tano-systems.com>
Date: Fri, 26 Feb 2021 03:40:57 +0300
Subject: [PATCH 3/4] ls1028a: Allow to defined CONFIG_BOOTCOMMAND in defconfig

Signed-off-by: Anton Kikin <a.kikin@tano-systems.com>
---
 include/configs/ls1028a_common.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/configs/ls1028a_common.h b/include/configs/ls1028a_common.h
index 9de5959290..4d3e0b08f0 100644
--- a/include/configs/ls1028a_common.h
+++ b/include/configs/ls1028a_common.h
@@ -190,7 +190,7 @@
 		"mmc dev 1;mmcinfo;mmc read $load_addr 0x4800 0x200 "		\
 		"&& hdp load $load_addr 0x2000\0"
 
-#undef CONFIG_BOOTCOMMAND
+#ifndef CONFIG_BOOTCOMMAND
 #if defined(CONFIG_SD_BOOT)
 #define CONFIG_BOOTCOMMAND "run sd_hdploadcmd; run distro_bootcmd;run sd_bootcmd; "	\
 			   "env exists secureboot && esbc_halt;"
@@ -203,6 +203,7 @@
 #endif
 #endif
 #endif
+#endif
 
 /* Monitor Command Prompt */
 #define CONFIG_SYS_CBSIZE		512	/* Console I/O Buffer Size */
-- 
2.17.1

