diff --git a/luasrc/controller/ddns.lua b/luasrc/controller/ddns.lua
index 24735289e..0e36edcb0 100755
--- a/luasrc/controller/ddns.lua
+++ b/luasrc/controller/ddns.lua
@@ -112,12 +112,8 @@ function service_version()
 	local srv_ver_cmd = luci_helper .. " -V | awk {'print $2'} "
 	local ver
 
-	if IPKG then
-		ver = IPKG.info(srv_name)[srv_name].Version
-	else
-		ver = UTIL.exec(srv_ver_cmd)
-	end
-	
+	ver = UTIL.exec(srv_ver_cmd)
+
 	if ver and #ver > 0 then return ver or nil end
 	
 end
