From efa6f064c33539c44ad9b0f87c4b8c73d6396279 Mon Sep 17 00:00:00 2001
From: Anton Kikin <a.kikin@tano-systems.com>
Date: Thu, 4 Jul 2019 04:22:17 +0300
Subject: [PATCH] Fix building for GCC 8.2

Signed-off-by: Anton Kikin <a.kikin@tano-systems.com>
---
 src/liboping.c | 27 +++++++++++++++++++--------
 1 file changed, 19 insertions(+), 8 deletions(-)

diff --git a/src/liboping.c b/src/liboping.c
index d99129c..61d8a45 100644
--- a/src/liboping.c
+++ b/src/liboping.c
@@ -198,8 +198,17 @@ static char *sstrerror (int errnum, char *buf, size_t buflen)
 static void ping_set_error (pingobj_t *obj, const char *function,
 	       	const char *message)
 {
+#if __GNUC__ >= 8
+/* GCC 7+ warns about the following calls that truncate a string using */
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-truncation"
+#endif
 	snprintf (obj->errmsg, sizeof (obj->errmsg),
 			"%s: %s", function, message);
+#if __GNUC__ >= 7
+#pragma GCC diagnostic pop
+#endif
+
 	obj->errmsg[sizeof (obj->errmsg) - 1] = 0;
 }
 
@@ -833,11 +842,12 @@ static int ping_send_one_ipv4 (pingobj_t *obj, pinghost_t *ph)
 	icmp4->icmp_id    = htons (ph->ident);
 	icmp4->icmp_seq   = htons (ph->sequence);
 
-	buflen = 4096 - sizeof (struct icmp);
-	strncpy (data, ph->data, buflen);
-	datalen = strlen (data);
+	datalen = strlen(ph->data);
+	buflen = sizeof (struct icmp) + datalen;
+	if (sizeof (buf) < buflen)
+		return (EINVAL);
 
-	buflen = datalen + sizeof (struct icmp);
+	memcpy (data, ph->data, datalen);
 
 	icmp4->icmp_cksum = ping_icmp4_checksum (buf, buflen);
 
@@ -880,11 +890,12 @@ static int ping_send_one_ipv6 (pingobj_t *obj, pinghost_t *ph)
 	icmp6->icmp6_id    = htons (ph->ident);
 	icmp6->icmp6_seq   = htons (ph->sequence);
 
-	buflen = 4096 - sizeof (struct icmp6_hdr);
-	strncpy (data, ph->data, buflen);
-	datalen = strlen (data);
+	datalen = strlen(ph->data);
+	buflen = sizeof (struct icmp6_hdr) + datalen;
+	if (sizeof (buf) < buflen)
+		return (EINVAL);
 
-	buflen = datalen + sizeof (struct icmp6_hdr);
+	memcpy (data, ph->data, datalen);
 
 	dprintf ("Sending ICMPv6 package with ID 0x%04x\n", ph->ident);
 
-- 
2.16.2.windows.1

