From ba7adc4eb6aac996e2dd90efe41d3d46776f7d26 Mon Sep 17 00:00:00 2001
From: Anton Kikin <a.kikin@tano-systems.com>
Date: Thu, 8 Dec 2022 10:16:27 +0300
Subject: [PATCH] Fix building for GCC 11

Signed-off-by: Anton Kikin <a.kikin@tano-systems.com>
---
 src/liboping.c | 8 ++++++++
 src/oping.c    | 3 +--
 2 files changed, 9 insertions(+), 2 deletions(-)

diff --git a/src/liboping.c b/src/liboping.c
index eca5c16..a690102 100644
--- a/src/liboping.c
+++ b/src/liboping.c
@@ -203,8 +203,16 @@ static char *sstrerror (int errnum, char *buf, size_t buflen)
 static void ping_set_error (pingobj_t *obj, const char *function,
 	       	const char *message)
 {
+#if __GNUC__ >= 8
+/* GCC 7+ warns about the following calls that truncate a string using */
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-truncation"
+#endif
 	snprintf (obj->errmsg, sizeof (obj->errmsg),
 			"%s: %s", function, message);
+#if __GNUC__ >= 7
+#pragma GCC diagnostic pop
+#endif
 	obj->errmsg[sizeof (obj->errmsg) - 1] = 0;
 }
 
diff --git a/src/oping.c b/src/oping.c
index 87ca81a..5fb0888 100644
--- a/src/oping.c
+++ b/src/oping.c
@@ -1600,8 +1600,7 @@ static void update_host_hook (pingobj_iter_t *iter, /* {{{ */
 
 			HOST_PRINTF ("%zu bytes from %s (%s): icmp_seq=%u ttl=%i ",
 					data_len, context->host, context->addr,
-					sequence, recv_ttl,
-					format_qos (recv_qos, recv_qos_str, sizeof (recv_qos_str)));
+					sequence, recv_ttl);
 			if ((recv_qos != 0) || (opt_send_qos != 0))
 			{
 				HOST_PRINTF ("qos=%s ",
-- 
2.34.1.windows.1

