From 7f1ea2f1ddd779e5cd6484970cf8d403af6497f7 Mon Sep 17 00:00:00 2001
From: Anton Kikin <a.kikin@tano-systems.com>
Date: Fri, 27 May 2022 05:38:21 +0300
Subject: [PATCH 1005/1013] rockchip: Do not parse bootargs from FDT

Signed-off-by: Anton Kikin <a.kikin@tano-systems.com>
---
 arch/arm/mach-rockchip/board.c | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/arch/arm/mach-rockchip/board.c b/arch/arm/mach-rockchip/board.c
index 587f4fd..c9fb2b3 100644
--- a/arch/arm/mach-rockchip/board.c
+++ b/arch/arm/mach-rockchip/board.c
@@ -1016,11 +1016,11 @@ void board_quiesce_devices(void *images)
 char *board_fdt_chosen_bootargs(void *fdt)
 {
 	/* bootargs_ext is used when dtbo is applied. */
-	const char *arr_bootargs[] = { "bootargs", "bootargs_ext" };
+	/*const char *arr_bootargs[] = { "bootargs", "bootargs_ext" };*/
 	const char *bootargs;
 	int nodeoffset;
-	int i, dump;
-	char *msg = "kernel";
+	int /*i,*/ dump;
+	/*char *msg = "kernel";*/
 
 	/* debug */
 	hotkey_run(HK_INITCALL);
@@ -1033,6 +1033,8 @@ char *board_fdt_chosen_bootargs(void *fdt)
 	if (nodeoffset < 0)
 		return NULL;
 
+/* Disabled by Tano Systems. Not required for SWUpdate */
+#if 0
 	for (i = 0; i < ARRAY_SIZE(arr_bootargs); i++) {
 		bootargs = fdt_getprop(fdt, nodeoffset, arr_bootargs[i], NULL);
 		if (!bootargs)
@@ -1061,6 +1063,7 @@ char *board_fdt_chosen_bootargs(void *fdt)
 		env_update("bootargs", bootargs);
 #endif
 	}
+#endif
 
 #ifdef CONFIG_MTD_BLK
 	char *mtd_par_info = mtd_part_parse(NULL);
-- 
2.34.1.windows.1

