From 80cf1846686f9f906095fb6d3aada988036b9cf7 Mon Sep 17 00:00:00 2001
From: Anton Kikin <a.kikin@tano-systems.com>
Date: Mon, 25 Jun 2018 21:50:28 +0300
Subject: [PATCH 3/4] Add 'find' command to opkg

---
 libopkg/opkg_cmd.c | 21 ++++++++++++++++++---
 src/opkg.c         |  1 +
 2 files changed, 19 insertions(+), 3 deletions(-)

diff --git a/libopkg/opkg_cmd.c b/libopkg/opkg_cmd.c
index 12e33a0..5fc9680 100644
--- a/libopkg/opkg_cmd.c
+++ b/libopkg/opkg_cmd.c
@@ -479,12 +479,13 @@ static int opkg_download_cmd(int argc, char **argv)
     return err;
 }
 
-static int opkg_list_cmd(int argc, char **argv)
+static int opkg_list_find_cmd(int argc, char **argv, int use_desc)
 {
-    unsigned int i;
+    int i;
     pkg_vec_t *available;
     pkg_t *pkg;
     char *pkg_name = NULL;
+    char *description;
 
     if (argc > 0) {
         pkg_name = argv[0];
@@ -494,8 +495,11 @@ static int opkg_list_cmd(int argc, char **argv)
     pkg_vec_sort(available, pkg_compare_names);
     for (i = 0; i < available->len; i++) {
         pkg = available->pkgs[i];
+        description = use_desc ? pkg->description : NULL;
         /* if we have package name or pattern and pkg does not match, then skip it */
-        if (pkg_name && fnmatch(pkg_name, pkg->name, opkg_config->nocase))
+        if (pkg_name && fnmatch(pkg_name, pkg->name, opkg_config->nocase) &&
+            (!use_desc || !description
+             || fnmatch(pkg_name, description, opkg_config->nocase)))
             continue;
         print_pkg(pkg);
     }
@@ -504,6 +508,16 @@ static int opkg_list_cmd(int argc, char **argv)
     return 0;
 }
 
+static int opkg_list_cmd(int argc, char **argv)
+{
+    return opkg_list_find_cmd(argc, argv, 0);
+}
+
+static int opkg_find_cmd(int argc, char **argv)
+{
+    return opkg_list_find_cmd(argc, argv, 1);
+}
+
 static int opkg_list_installed_cmd(int argc, char **argv)
 {
     unsigned int i;
@@ -1107,6 +1121,7 @@ static opkg_cmd_t cmds[] = {
     {"files", 1, (opkg_cmd_fun_t) opkg_files_cmd, PFM_DESCRIPTION | PFM_SOURCE},
     {"search", 1, (opkg_cmd_fun_t) opkg_search_cmd,
         PFM_DESCRIPTION | PFM_SOURCE},
+    {"find", 1, (opkg_cmd_fun_t) opkg_find_cmd, PFM_SOURCE},
     {"download", 1, (opkg_cmd_fun_t) opkg_download_cmd,
         PFM_DESCRIPTION | PFM_SOURCE},
     {"compare_versions", 1, (opkg_cmd_fun_t) opkg_compare_versions_cmd, 0},
diff --git a/src/opkg.c b/src/opkg.c
index 2ead642..71a55e6 100644
--- a/src/opkg.c
+++ b/src/opkg.c
@@ -280,6 +280,7 @@ static void usage()
     printf("\tlist-changed-conffiles          List user modified configuration files\n");
     printf("\tfiles <pkg>                     List files belonging to <pkg>\n");
     printf("\tsearch <file|glob>              List package providing <file>\n");
+    printf("\tfind <regexp>                   List packages whose name or description matches <regexp>\n");
     printf("\tinfo [pkg|glob]                 Display all info for <pkg>\n");
     printf("\tstatus [pkg|glob]               Display all status for <pkg>\n");
     printf("\tdownload <pkg>                  Download <pkg> to current directory\n");
-- 
2.7.4

