From 95af9e161855852788963c41b01366e1f57a7abb Mon Sep 17 00:00:00 2001
From: Anton Kikin <a.kikin@tano-systems.com>
Date: Thu, 15 Nov 2018 16:23:23 +0300
Subject: [PATCH 5/9] block: Fix some memory leaks and unnecessary duplicate
 mounts

Signed-off-by: Anton Kikin <a.kikin@tano-systems.com>
---
 block.c | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/block.c b/block.c
index 41dd4c6..94483e4 100644
--- a/block.c
+++ b/block.c
@@ -1055,7 +1055,22 @@ static int mount_device(struct probe_info *pr, int type)
 
 	m = find_block(pr->uuid, pr->label, device, NULL);
 	if (m && m->extroot)
+	{
+		if (mp)
+			free(mp);
 		return -1;
+	}
+
+	if (m && mp && (strcmp(mp, m->target) == 0))
+	{
+		ULOG_INFO("%s is already mounted on requested target %s\n", pr->dev, m->target);
+		free(mp);
+		return 0;
+	}
+
+	/* mp is no longer used */
+	if (mp)
+		free(mp);
 
 	if (type == TYPE_HOTPLUG)
 		blockd_notify(device, m, pr);
-- 
2.16.2.windows.1

