From 09c4a74b26d007555caa51014e969868b3a420a7 Mon Sep 17 00:00:00 2001
From: Anton Kikin <a.kikin@tano-systems.com>
Date: Thu, 15 Nov 2018 16:24:21 +0300
Subject: [PATCH 06/12] blockd: Fix simultaneous multiple devices mounts
 handling

Signed-off-by: Anton Kikin <a.kikin@tano-systems.com>
---
 blockd.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/blockd.c b/blockd.c
index e07530c..f7a7cd4 100644
--- a/blockd.c
+++ b/blockd.c
@@ -237,6 +237,7 @@ block_hotplug(struct ubus_context *ctx, struct ubus_object *obj,
 	if (!device)
 		return UBUS_STATUS_UNKNOWN_ERROR;
 
+	/* vlist_update(&devices); */
 	if (data[MOUNT_REMOVE]) {
 		vlist_delete(&devices, &device->node);
 	} else {
-- 
2.16.2.windows.1

