From 97d593c796560328c4f96d32fdbc7f2a08cfb4fa Mon Sep 17 00:00:00 2001
From: Anton Kikin <a.kikin@tano-systems.com>
Date: Wed, 3 Jun 2020 06:00:27 +0300
Subject: [PATCH 4/8] musb_core: Add sysfs entry for musb->port1_status value

Signed-off-by: Anton Kikin <a.kikin@tano-systems.com>
---
 drivers/usb/musb/musb_core.c | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/drivers/usb/musb/musb_core.c b/drivers/usb/musb/musb_core.c
index bf083c1f997f..0158f1bd1908 100644
--- a/drivers/usb/musb/musb_core.c
+++ b/drivers/usb/musb/musb_core.c
@@ -1829,10 +1829,26 @@ static ssize_t srp_store(struct device *dev, struct device_attribute *attr,
 }
 static DEVICE_ATTR_WO(srp);
 
+static ssize_t port1_status_show(struct device *dev,
+		struct device_attribute *attr, char *buf)
+{
+	struct musb  *musb = dev_to_musb(dev);
+	unsigned int  status;
+	unsigned long flags;
+
+	spin_lock_irqsave(&musb->lock, flags);
+	status = musb->port1_status;
+	spin_unlock_irqrestore(&musb->lock, flags);
+
+	return sprintf(buf, "0x%x\n", status);
+}
+static DEVICE_ATTR_RO(port1_status);
+
 static struct attribute *musb_attrs[] = {
 	&dev_attr_mode.attr,
 	&dev_attr_vbus.attr,
 	&dev_attr_srp.attr,
+	&dev_attr_port1_status.attr,
 	NULL
 };
 ATTRIBUTE_GROUPS(musb);
-- 
2.17.1

