From 37ee7e59b298d20e2c35854e99ae6e02bfc93804 Mon Sep 17 00:00:00 2001
From: Anton Kikin <a.a.kikin@gmail.com>
Date: Tue, 24 Oct 2017 01:09:41 +0300
Subject: [PATCH] Startup delay patch.

---
 utils/ifupdown.sh.in | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/utils/ifupdown.sh.in b/utils/ifupdown.sh.in
index ededf05..e217a70 100755
--- a/utils/ifupdown.sh.in
+++ b/utils/ifupdown.sh.in
@@ -96,6 +96,8 @@ if [ "$MODE" = 'start' ]; then
   # This triggers the kernel to run `/sbin/bridge-stp start $IFACE`
   "$brctl" stp "$IFACE" on
 
+  sleep 1
+
   # `mstpctl addbridge $IFACE` must be called before this script continues.
   # If mstpd is already running then /sbin/bridge-stp will call
   # `mstpctl addbridge $IFACE` before `brctl stp $IFACE on` returns.
-- 
2.7.4

