From b94c54a6b8106ac13c2eb8b2d34099697331ecf5 Mon Sep 17 00:00:00 2001
From: Anton Kikin <a.kikin@tano-systems.com>
Date: Thu, 4 Jul 2019 05:01:51 +0300
Subject: [PATCH 2/6] Fix building for GCC 8.2

Signed-off-by: Anton Kikin <a.kikin@tano-systems.com>
---
 file.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/file.c b/file.c
index e6282aa..9edd14c 100644
--- a/file.c
+++ b/file.c
@@ -147,6 +147,7 @@ uh_path_lookup(struct client *cl, const char *url)
 
 	int i = 0;
 	int len;
+	int retval;
 	struct stat s;
 	struct index_file *idx;
 
@@ -202,7 +203,9 @@ uh_path_lookup(struct client *cl, const char *url)
 		if (!exists)
 			continue;
 
-		snprintf(path_gz, sizeof(path_gz), "%s.gz", path_phys);
+		retval = snprintf(path_gz, sizeof(path_gz), "%s.gz", path_phys);
+		if ((retval < 0) || (retval >= sizeof(path_gz)))
+			return NULL;
 		
 		/* test both ordinary path and gz path */
 		if (stat(path_phys, &p.stat) == 0){
-- 
2.28.0.windows.1

