From 68f9657c3d8d47cfe514b2bf0283e6c53ada0b98 Mon Sep 17 00:00:00 2001
From: Anton Kikin <a.kikin@tano-systems.com>
Date: Sun, 10 Jan 2021 16:24:21 +0300
Subject: [PATCH 1004/1006] main: Fix formatting in usage help text

Signed-off-by: Anton Kikin <a.kikin@tano-systems.com>
---
 main.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/main.c b/main.c
index 694278e..e36c01a 100644
--- a/main.c
+++ b/main.c
@@ -163,8 +163,8 @@ static int usage(const char *name)
 		"	-u string       URL prefix for UBUS via JSON-RPC handler\n"
 		"	-U file         Override ubus socket path\n"
 		"	-a              Do not authenticate JSON-RPC requests against UBUS session api\n"
-		"	-X		Enable CORS HTTP headers on JSON-RPC api\n"
-		"	-e		Events subscription reconnection time (retry value)\n"
+		"	-X              Enable CORS HTTP headers on JSON-RPC api\n"
+		"	-e milliseconds Events subscription reconnection time (retry value)\n"
 #endif
 		"	-x string       URL prefix for CGI handler, default is '/cgi-bin'\n"
 		"	-y alias[=path]	URL alias handle\n"
-- 
2.34.1.windows.1

