Index: net-snmp-5.7.3/agent/mibgroup/if-mib/data_access/interface_linux.c
===================================================================
--- net-snmp-5.7.3.orig/agent/mibgroup/if-mib/data_access/interface_linux.c
+++ net-snmp-5.7.3/agent/mibgroup/if-mib/data_access/interface_linux.c
@@ -16,6 +16,8 @@ netsnmp_feature_child_of(interface_arch_
 netsnmp_feature_require(interface_ioctl_flags_set)
 #endif /* NETSNMP_FEATURE_REQUIRE_INTERFACE_ARCH_SET_ADMIN_STATUS */
 
+#undef HAVE_PCI_LOOKUP_NAME
+#undef HAVE_PCI_PCI_H
 #ifdef HAVE_PCI_LOOKUP_NAME
 #include <pci/pci.h>
 #include <setjmp.h>
