From 3e90aed24d7c5f41ad6ef632f1b8333ad6ad8720 Mon Sep 17 00:00:00 2001
From: Anton Kikin <a.kikin@tano-systems.com>
Date: Mon, 15 Mar 2021 20:10:05 +0300
Subject: [PATCH 201/201] ti-am574x: Adjust SYS_MMC_ENV_DEV for SD/eMMC boot

Signed-off-by: Anton Kikin <a.kikin@tano-systems.com>
---
 include/configs/ti_armv7_common.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/include/configs/ti_armv7_common.h b/include/configs/ti_armv7_common.h
index 08829781e3..bee151facf 100644
--- a/include/configs/ti_armv7_common.h
+++ b/include/configs/ti_armv7_common.h
@@ -178,6 +178,12 @@
 #define CONFIG_SYS_MMCSD_RAW_MODE_ARGS_SECTORS	0x200   /* 256KiB */
 #endif
 
+#if defined(CONFIG_EMMC_BOOT)
+#define CONFIG_SYS_MMC_ENV_DEV		1
+#else
+#define CONFIG_SYS_MMC_ENV_DEV		0
+#endif
+
 /* General parts of the framework, required. */
 
 #ifdef CONFIG_MTD_RAW_NAND
-- 
2.25.1

